components/textbook-beta/DynamicBackgroundLogo.vue

```diff
@@ -0,0 +1,251 @@
+      lastRender: 0
+    }
+  },
+  computed: {
+    maskHeight: () : number => imgMask.length,
+    maskWidth: () : number => imgMask[0].length,
+    parent () : HTMLElement { return this.$refs.canvasWrapper as HTMLElement },
+    canvas () : HTMLCanvasElement { return this.$refs.canvas as HTMLCanvasElement },
+    renderingContext () : CanvasRenderingContext2D { return this.canvas.getContext('2d')! }
+  },
+  mounted () {
+    this.initRenderLoop()
+  },
+  methods: {
+    initRenderLoop () {
+      this.renderingContext.globalCompositeOperation = 'destination-atop'
+
+      this.drawSpeed = this.initialDrawSpeed * DRAWSPEED_CORRECTION
+
+      window.addEventListener('resize', this.resize)
+      window.addEventListener('keydown', this.interactionRedraw)
+      window.addEventListener('touchstart', this.interactionRedraw)
+      this.resize()
+
+      this.randomizeMask()
+      window.requestAnimationFrame(this.loop)
+    },
+    loop (timestamp : DOMHighResTimeStamp) {
+      if (this.lastRender === 0) {
+        this.lastRender = timestamp
+      }
+      const deltaTime = Math.min(timestamp - this.lastRender, 20)
+
+      this.draw(deltaTime)
+
+      this.lastRender = timestamp
+      window.requestAnimationFrame(this.loop)
+    },
+    draw (deltaTime: number) {
+      const deltaTimeSeconds = deltaTime * 0.001
+      const frameProgress = deltaTimeSeconds * this.drawSpeed
+      const newAnimationProgress = Math.min(this.animationProgress + frameProgress, 1)
+      // prevent redrawing on finished animation
+      if (newAnimationProgress === this.animationProgress) {
+        return
+      }
+      this.animationProgress = newAnimationProgress
+      this.renderingContext!.clearRect(0, 0, this.canvas!.width, this.canvas!.height)
+
+      for (let i = 0; i < this.maskWidth; i++) {
+        for (let j = 0; j < this.maskHeight; j++) {
+          const gridElement = this.gridData[j][i]
+          if (gridElement.value) {
+            const alpha = Math.pow(this.clamp(this.animationProgress * 2 - gridElement.delay), 2) * 2
+            this.renderingContext!.strokeStyle = `rgba(255, 255, 255, ${alpha})`
+            const finalSize = this.squareSize / gridElement.value
+            this.renderingContext!.strokeRect(SPACING_BETWEEN_CELLS + i * this.squareSizeWithSpacing, SPACING_BETWEEN_CELLS + j * this.squareSizeWithSpacing, finalSize, finalSize)
+          }
+        }
+      }
+    },
+    resize () {
+      this.parentWidth = this.parent!.clientWidth
+      this.parentHeight = this.parent!.clientHeight
+      this.squareSizeWithSpacing = (this.parentWidth - 2 * SPACING_BETWEEN_CELLS) / Math.max(this.maskHeight, this.maskWidth)
+      this.squareSize = this.squareSizeWithSpacing - SPACING_BETWEEN_CELLS
+
+      const pixelDensityConsideringDevicePixelRatio = (window.devicePixelRatio || 1) * this.pixelDensity
+
+      // this disparity of html-width/height and css-width/height
+      // creates a high resolution canvas with better lines
```

Comment:
A multiple-line comment should use the `/* */` style [per spec](https://tc39.es/ecma262/#prod-MultiLineComment). `//` is for single line comments.